pkg/api/hosttwin.go

```diff
@@ -0,0 +1,62 @@
+package api
+
+import (
+	"fmt"
+
+	"github.com/Iotic-Labs/iotics-identity-go/pkg/advancedapi"
+	"github.com/Iotic-Labs/iotics-identity-go/pkg/register"
+)
+
+// DelegateControlByPrivateExponent registers a twin identity with a control delegate to agent
+func DelegateControlByPrivateExponent(resolverClient register.ResolverClient, twinIssuer *register.Issuer, twinPrivateExponent string, agentID register.RegisteredIdentity, delegationName string) error {
+	twinKeypair, err := advancedapi.GetKeyPairFromPrivateExponentHex(twinPrivateExponent)
+	if err != nil {
+		return err
+	}
+
+	twinID := register.NewRegisteredIdentity(twinKeypair, twinIssuer)
+
+	return TwinDelegatesControlToAgent(resolverClient, twinID, agentID, delegationName)
+}
+
+// GetOwnershipOfTwinByPrivateExponent Get Ownership of a twin using the private exponent of the twin.
+func GetOwnershipOfTwinByPrivateExponent(resolverClient register.ResolverClient, twinIssuer *register.Issuer, twinPrivateExponent string, newOwnerID register.RegisteredIdentity, newOwnerKeyName string) error {
+	twinKeypair, err := advancedapi.GetKeyPairFromPrivateExponentHex(twinPrivateExponent)
+	if err != nil {
+		return err
+	}
+
+	twinID := register.NewRegisteredIdentity(twinKeypair, twinIssuer)
+
+	return advancedapi.AddPublicKeyToDocument(resolverClient, nil, newOwnerKeyName, newOwnerID.KeyPair().PublicKeyBase58, twinID)
+}
+
+// GetOwnershipOfTwinAndDelegateControlByPrivateExponent Get Ownership of a twin using the agent identity and delegate control to that agent using twin private key exponent.
+func GetOwnershipOfTwinAndDelegateControlByPrivateExponent(resolverClient register.ResolverClient, twinIssuer *register.Issuer, twinPrivateExponent string, agentID register.RegisteredIdentity, newOwnerKeyName string, delegationName string) error {
+	twinKeypair, err := advancedapi.GetKeyPairFromPrivateExponentHex(twinPrivateExponent)
+	if err != nil {
+		return err
+	}
+
+	twinDoc, err := resolverClient.GetDocument(twinIssuer.Did)
+	if err != nil {
+		return err
+	}
+
+	dProof, err := advancedapi.CreateProof(agentID.KeyPair(), agentID.Issuer(), []byte(twinIssuer.Did))
+	if err != nil {
+		return err
+	}
+
+	opts := []register.RegisterDocumentOpts{
+		register.AddFromExistingDocument(twinDoc),
+		register.AddPublicKey(newOwnerKeyName, agentID.KeyPair().PublicKeyBase58, false),
```

Comment:
Is this not `newOwnerID.KeyPair().PublicKeyBase58` like in above func?

Reply:
Thanks, fixed